src/gui/addnewtorrentdialog.cpp

```diff
@@ -279,13 +279,20 @@ void AddNewTorrentDialog::saveSavePathHistory() const
     // Get current history
     QStringList history = pref->getAddNewTorrentDialogPathHistory();
     QList<QDir> history_dirs;
-    foreach(const QString dir, history)
-        history_dirs << QDir(dir);
+    int defaultSavelistHistorySize = pref->getSavelistHistorySize();
```

Comment:
IIRC, the paths are saved sorted by date(can't remember which order). So check if the list size exceeds the configured one. If it does take the appropriate the most recent ones and discards the others. No need to iterate.